### PR TITLE
Added authorizations requirements for users and tasks

### PR DESCRIPTION
**Pull request by ffouillet**
- Only users with role ROLE_ADMIN should be able to access users management actions (create, delete)
- Tasks should be deleted only by their authors.
- Tasks without an author should only be deleted by users with role ROLE_ADMIN

This branch have been created on top of "dev_add_user_roles" branch.

### CODE REVIEW

File: src/Controller/TaskController.php [hunk 1]
```diff
@@ -45,13 +49,18 @@ public function createAction(Request $request)
...
+
+        $form = $this->createForm(TaskEditType::class, $task);
 
         $form->handleRequest($request);
 
-        if ($form->isValid()) {
+        if ($form->isSubmitted() && $form->isValid()) {
+
+            // Only task author can edit his task. Users with ROLE_ADMIN can edit all tasks.
+            $this->denyAccessUnlessGranted('edit', $task, "Désolé, seul les auteurs de leur tâches peuvent les modifier.");
```
Comment (olix21) [inline src/Controller/TaskController.php]: > Désolé, seulS les auteurs de leurS tâches

Mais dans ce cas-ci, ça serait plutôt

> Désolé, seul l'auteurs d'une tâche peut la modifier

File: src/Security/TaskVoter.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        throw new \LogicException('This code should not be reached!');
+    }
+
+    private function canDelete(Task $task, User $authenticatedUser)
+    {
+        // Task should be delete only by their authors.
+        // Else if task has no author, task should only be deletable by an user with ROLE_ADMIN
+        if($task->getAuthor() !== null && ($task->getAuthor()->getId() == $authenticatedUser->getId()) ||
+            ($task->getAuthor() == null && $this->security->isGranted('ROLE_ADMIN'))) {
```
Comment (olix21) [inline src/Security/TaskVoter.php]: It's correct to do like this. I'm just wondering what performance gain we can have if we:
- Don't inject the `Security` service
- add an `hasRole` method to the user and use it here (as we already have the authenticated user)

**Review by olix21 — APPROVED** [review]
Good work. Just wondering about some minor (to not say invisible) perf optimizations

**Comment by olix21** [comment]
don't merge this until you have made the changes in the previous PR

File: src/Controller/TaskController.php (hunk 1 above)
Comment (ffouillet) [reply src/Controller/TaskController.php]: I agree with this.
Corrected in b69a8d8.

File: src/Security/TaskVoter.php (hunk 2 above)
Comment (ffouillet) [reply src/Security/TaskVoter.php]: I find your solution more elegant and will use it, thanks for the ip ! And as you said, it'll avoid us to inject security service... I'll also apply this to the UserVoter.

Comment (ffouillet) [reply src/Security/TaskVoter.php]: Correction applied in d279522

**Pull request merged**